src/tools/illink/test/Mono.Linker.Tests/Mono.Linker.Tests.csproj

```diff
@@ -18,30 +17,25 @@
     <PackageReference Include="Microsoft.DotNet.Cecil.Pdb" Version="$(MicrosoftDotNetCecilPdbVersion)" PrivateAssets="All" Publish="True" />
     <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="$(MicrosoftCodeAnalysisVersion)" />
     <ProjectReference Include="$(RepoRoot)\src\coreclr\tools\ILVerification\ILVerification.csproj" />
-    <PackageReference Include="nunit" Version="3.12.0" />
-    <PackageReference Include="NUnit3TestAdapter" Version="4.1.0" />
+    <PackageReference Include="nunit" Version="$(NUnitVersion)" />
+    <PackageReference Include="NUnit3TestAdapter" Version="$(NUnitTestAdapterVersion)" />
     <!-- This reference is purely so that the linker can resolve this
          dependency of mscorlib. It is not actually required to build
          the tests. -->
-    <PackageReference Include="System.Threading.AccessControl" Version="5.0.0" />
+    <PackageReference Include="System.Threading.AccessControl" Version="$(SystemThreadingAccessControlVersion)" />
 
     <ProjectReference Include="..\..\src\linker\Mono.Linker.csproj" SkipUseReferenceAssembly="true" />
     <ProjectReference Include="..\Mono.Linker.Tests.Cases\Mono.Linker.Tests.Cases.csproj" />
     <ProjectReference Include="..\Mono.Linker.Tests.Cases.Expectations\Mono.Linker.Tests.Cases.Expectations.csproj" />
   </ItemGroup>
 
-  <Target Name="PrepareTools" BeforeTargets="Build">
-    <!-- Restore ilasm using ilasm.ilproj. Restore must be done
-         separately from copy with a different set of input
-         properties, to force MSBuild to re-evaluate using new props
-         and targets from the restored package. -->
-    <PropertyGroup>
-      <IlasmProject>$(MSBuildThisFileDirectory)../../ilasm.ilproj</IlasmProject>
-    </PropertyGroup>
-
-    <MSBuild Projects="$(IlasmProject)" Targets="Restore" />
-    <MSBuild Projects="$(IlasmProject)" Targets="CopyILAsmTool" />
-  </Target>
+  <!-- Tests require ilasm so we make sure that it is live built before executing the tests. -->
+  <ItemGroup>
+    <ProjectReference Include="$(CoreClrProjectRoot)runtime.proj"
+                      AdditionalProperties="ClrILToolsSubset=true;Configuration=$(CoreCLRConfiguration)"
+                      ReferenceOutputAssembly="false"
+                      SkipGetTargetFrameworkProperties="true" />
+  </ItemGroup>
```

Comment:
As mentioned offline, this will likely lead to really difficult to diagnose build issues when trying to build any of the native components and this test project in the same build invocation (such as `clr+libs+tools.illink+tools.illinktests` in one build) due to there being two native builds of the same code with the same output directory running simultaneously.

Reply:
> due to there being two native builds of the same code with the same output directory running simultaneously.

We don't build subsets in parallel. How would your above example result in two native builds running simultaneously?

Reply:
I thought we had changed some of the subsets to build in parallel. Or was that just the libraries projects that used to have project references that used to cross between the subsets (like a src project referencing a ref project when our subsets were libs.ref and libs.src)?

Reply:
While a few individual subsets build projects in parallel, the majority doesn't: https://github.com/dotnet/runtime/blob/e395c7c767437b9b8d10113bb8fc816ffb30f521/eng/Subsets.props#L192.

Inside the libs subset, we have a build sequence as well:
1. libs.native
2. libs.sfx
3. libs.oob
4. libs.pretest
5. (optionally: libs.tests)

The projects in those subsets build in parallel but the subsets themselves run in sequential order. Also, by having a ProjectReference to runtime.proj, we guarantee that the project doesn't build multiple times (as long as the same set of global properties are passed in).

Reply:
The problem with the ProjectReference model is that the global properties won't match if any clr subset that builds native code other than `clr.iltools` is specified. So, we'll at best re-enter the native build script and then not do any additional work.

Since we run subsets sequentially, we won't have the simultaneous-build problem, so it's not the worst.

Reply:
> So, we'll at best re-enter the native build script and then not do any additional work.
Since we run subsets sequentially, we won't have the simultaneous-build problem, so it's not the worst.

Yes I tried that out and the incremental build takes less than five seconds. The double evaluation of runtime.proj is neglectable as such are in the 10-100ms range.

> The problem with the ProjectReference model is that the global properties won't match if any clr subset that builds native code other than clr.iltools is specified.

That's correct but IMO fine for now. We eventually want to come up with a design that makes this work well for the entire repository. That might involve creating separate msbuild wrapper projects for the different components to avoid the global property mismatch. I haven't looked at what the build-runtime.cmd/sh scripts exactly do but it might also help to break that apart into logical parts, similar to how the corehost project only builds the host native files. Anyway that by large stretches what we are trying to achieve in this PR. Let's move that discussion over to https://github.com/dotnet/runtime/issues/43110.

Reply:
Sounds good